persistence/src/main/scala/com/lightbend/lagom/javadsl/persistence/cassandra/CassandraPersistentEntityRegistry.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.persistence.cassandra
+
+import java.util.{ Optional, UUID }
+
+import akka.NotUsed
+import akka.japi.Pair
+import akka.stream.javadsl
+import com.lightbend.lagom.javadsl.persistence.{ AggregateEvent, AggregateEventTag, PersistentEntityRegistry }
+
+/**
+ * Functionality specific to the Cassandra persistent entity registry.
+ */
+trait CassandraPersistentEntityRegistry extends PersistentEntityRegistry {
+
+  /**
+   * A stream of the persistent events that have the given `aggregateTag`, e.g.
+   * all persistent events of all `Order` entities.
+   */
+  def uuidEventStream[Event <: AggregateEvent[Event]](
+    fromOffset:   Optional[UUID],
```

Comment:
Sorry, I haven't been clear. It is very close to what you are suggesting, but with a different twist. Because you said that having an `Offset` type may not very intuitive, I thought we can avoid it altogether:
* `eventStream(offset: UUID, aggregateTag: AggregateEventTag[Event])`
* `eventStream(offset: Long, aggregateTag: AggregateEventTag[Event])`
* `eventStream(aggregateTag: AggregateEventTag[Event])` (this is the `NoOffset/BeginningOfUniverse ` case).

Having said that, I'd personally prefer to introduce an Offset type, as it makes the user code clearer. Especially in the case where no explicit offset is passed.